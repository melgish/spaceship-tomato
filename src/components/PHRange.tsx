type Props = Readonly<{
  range?: {
    maxPH: number
    minPH: number
  } | null
}>

export default function PHRange({ range }: Props) {
  if (range) {
    return (
      <>
        {range.minPH.toFixed(1)} &ndash; {range.maxPH.toFixed(1)}
      </>
    )
  }
}
