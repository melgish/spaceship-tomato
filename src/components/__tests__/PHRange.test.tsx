import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import PHRange from '@/components/PHRange'

const data = { maxPH: 8.22, minPH: 6.41 }

describe('when range is not supplied', () => {
  it('should return falsy value', () => {
    const { container } = render(<PHRange />)

    expect(container).toMatchSnapshot()
  })
})

describe('when range is valid', () => {
  it('should render range', () => {
    const { container } = render(<PHRange range={data} />)

    expect(container).toMatchSnapshot()
  })
})
