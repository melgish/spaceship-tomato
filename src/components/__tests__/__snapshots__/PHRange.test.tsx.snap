// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when range is not supplied > should return falsy value 1`] = `<div />`;

exports[`when range is valid > should render range 1`] = `
<div>
  6.4
   â€“ 
  8.2
</div>
`;
